torch/_decomp/decompositions.py

```diff
@@ -1247,3 +1247,98 @@ def fast_pow(x, ord):
         self = self.abs()
 
     return fast_pow(fast_pow(self, p).sum(dim, keepdim=keepdim), 1.0 / p)
+
+
+@register_decomposition(torch.ops.aten.upsample_bilinear2d.vec)
+def upsample_bilinear2d_vec(
+    input: Tensor,
+    output_size: Optional[List[int]],
+    align_corners: bool,
+    scale_factors: Optional[List[float]],
+) -> Tensor:
+    # get dimensions of original image
+    n_batch, n_channels, in_h, in_w = input.shape
+
+    if output_size is not None:
+        out_h = float(output_size[0])
+        out_w = float(output_size[1])
+    elif scale_factors is not None:
+        out_h = in_h * scale_factors[0]
+        out_w = in_w * scale_factors[1]
+
+    # Calculate horizontal and vertical scaling factor
+    if out_h > 1:
+        if align_corners:
+            h_scale_factor = (in_h - 1) / (int(out_h) - 1)
+        else:
+            h_scale_factor = in_h / out_h
+    else:
+        h_scale_factor = 0.0
+
+    if out_w > 1:
+        if align_corners:
+            w_scale_factor = (in_w - 1) / (int(out_w) - 1)
+        else:
+            w_scale_factor = in_w / out_w
+    else:
+        w_scale_factor = 0.0
+
+    i = torch.arange(out_h, dtype=input.dtype, device=input.device)
+    j = torch.arange(out_w, dtype=input.dtype, device=input.device)
+
+    if align_corners:
+        x = h_scale_factor * i
+        y = w_scale_factor * j
+    else:
+        x = (h_scale_factor * (i + 0.5) - 0.5).clamp(min=0.0)
+        y = (w_scale_factor * (j + 0.5) - 0.5).clamp(min=0.0)
+
+    x_floor = torch.floor(x)
+    x_ceil = torch.minimum(torch.ceil(x), torch.tensor(in_h - 1))
+    y_floor = torch.floor(y)
+    y_ceil = torch.minimum(torch.ceil(y), torch.tensor(in_w - 1))
+
+    x_view = x.view(1, 1, len(x), 1)
+    x_floor_view = x_floor.view(1, 1, len(x_floor), 1)
+    x_ceil_view = x_ceil.view(1, 1, len(x_ceil), 1)
```

Comment:
avoid view functions unless necessary, in this case enough to do 
```suggestion
    x_view = x.unsqueeze(-1)
    x_floor_view = x_floor.unsqueeze(-1)
    x_ceil_view = x_ceil.unsqueeze(-1)
```
the rest of the dimensions will be taken care of with implicit broadcast.